src/types/message_entity.rs

```diff
@@ -39,21 +39,90 @@ pub enum MessageEntityKind {
     Strikethrough,
 }
 
-#[test]
-fn recursive_kind() {
-    use serde_json::from_str;
+impl MessageEntity {
+    pub fn text_from(&self, message: Message) -> Option<String> {
```

Comment:
Why not `message: &Message`? Useful for processing many entities

Reply:
fixed